src/sys/windows/tcp.rs

```diff
@@ -832,6 +851,26 @@ impl fmt::Debug for TcpListener {
     }
 }
 
+impl FromRawSocket for TcpListener {
+    unsafe fn from_raw_socket(socket: RawSocket) -> TcpListener {
+        let stream = net::TcpListener::from_raw_socket(socket);
+        TcpListener::new(stream)
+            .expect("tried to get local address of invalid/unbound socket")
```

Comment:
This message seems to be incorrect.

Reply:
The only error that can possibly be returned when calling `self::TcpListener::new` is by a call to `std::net::TcpListener::local_addr`. [(code)](https://github.com/vikrrrr/mio/blob/as-raw-socket/src/sys/windows/tcp.rs#L667-L674)

It is true that it may be confusing, especially if `self::TcpListener::new` is ever modified to return another error.